From 530bb4635de6688fc97800aa84d4690460f36e8e Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:53:26 -0400
Subject: [tools] perf/symbols: Fix crash in elf_section_by_name

Message-id: <1408953210-26343-319-git-send-email-jolsa@redhat.com>
Patchwork-id: 88988
O-Subject: [PATCH RHEL7.1 BZ1133083 318/322] perf symbols: Fix crash in elf_section_by_name
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 155b3a13a65d4217316dbe094843f2a7df0711fa
Author: Jiri Olsa <jolsa@redhat.com>
Date: Sun Mar 2 14:32:07 2014 +0100

description
===========
Fixing crash in elf_section_by_name function caused by missing section
name in elf binary.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/symbol-elf.c b/tools/perf/util/symbol-elf.c
index 3e9f336..516d19f 100644
--- a/tools/perf/util/symbol-elf.c
+++ b/tools/perf/util/symbol-elf.c
@@ -151,15 +151,15 @@ Elf_Scn *elf_section_by_name(Elf *elf, GElf_Ehdr *ep,
 
   gelf_getshdr(sec, shp);
   str = elf_strptr(elf, ep->e_shstrndx, shp->sh_name);
-  if (!strcmp(name, str)) {
+  if (str && !strcmp(name, str)) {
    if (idx)
     *idx = cnt;
-   break;
+   return sec;
   }
   ++cnt;
  }
 
- return sec;
+ return NULL;
 }
 
 #define elf_section__for_each_rel(reldata, pos, pos_mem, idx, nr_entries) \
-- 
1.7.1